src/components/connect.js

```diff
@@ -17,9 +27,127 @@ function getDisplayName(WrappedComponent) {
+function connectFactory(defaults = {}) {
+  function connectImpl(map, options = {}) {
+    /* eslint-disable no-console */
+    if (Object.getOwnPropertyNames(options).length > 0 && console && console.warn) {
+      console.warn('The options argument is deprecated in favor of `connect.defaults()` calls. In a future release, support will be removed.')
+    }
+
+    const finalOptions = Object.assign({}, defaults, options)
+
+    if (Function.prototype.isPrototypeOf(finalOptions.buildRequest) &&
+      Function.prototype.isPrototypeOf(finalOptions.Request) &&
+      console && console.info) {
+      console.info('Both buildRequest and Request were provided in `connect.defaults()`. However this custom Request would only be used in the default buildRequest.')
+    }
+
+    /* eslint-enable no-console */
+    return connect(map, finalOptions)
+  }
+
+  connectImpl.defaults = function (overrides = {}) {
+    return connectFactory(Object.assign({}, defaults, overrides))
+  }
+
+  return connectImpl
+}
+
+export default connectFactory()
+
+function typecheck(type, name, obj) {
+  invariant(
+    typeof obj === type,
+    `${name} must be a ${type}. Instead received a %s.`,
+    typeof obj
+  )
+}
+
+const checks = {
+  buildRequest(fn) {
+    typecheck('function', 'buildRequest', fn)
+  },
+
+  credentials(str) {
+    const allowed = [ 'omit', 'same-origin', 'include' ]
+    invariant(
+      allowed.indexOf(str) !== -1,
+      `credentials must be one of ${allowed.join(', ')}. Instead got %s.`,
+      str ? str.toString() : str
+    )
+  },
+
+  fetch(fn) {
+    typecheck('function', 'fetch', fn)
+  },
+
+  handleResponse(fn) {
+    typecheck('function', 'handleResponse', fn)
+  },
+
+  headers(obj) {
+    invariant(
+      isPlainObject(obj),
+      'headers must be a plain object with string values. Instead received a %s.',
+      typeof obj
+    )
+  },
+
+  method(str) {
+    typecheck('string', 'method', str)
+  },
+
+  redirect(str) {
+    const allowed = [ 'follow', 'error', 'manual' ]
+    invariant(
+      allowed.indexOf(str) !== -1,
+      `redirect must be one of ${allowed.join(', ')}. Instead got %s.`,
+      str ? str.toString() : str
+    )
+  },
+
+  refreshInterval(num) {
+    typecheck('number', 'refreshInterval', num)
+    invariant(num >= 0, 'refreshInterval must be positive or 0.')
+    invariant(num !== Infinity, 'refreshInterval must not be Infinity.')
+  },
+
+  Request(fn) {
+    typecheck('function', 'Request', fn)
+  }
+}
+
+function connect(mapPropsToRequestsToProps, defaults) {
   const finalMapPropsToRequestsToProps = mapPropsToRequestsToProps || defaultMapPropsToRequestsToProps
-  const { withRef = false } = options
+
+  defaults = Object.assign({
+    andCatch: undefined,
+    andThen: undefined,
+    buildRequest,
+    catch: undefined,
+    comparison: undefined,
+    credentials: 'same-origin',
+    fetch: top.fetch,
+    force: false,
+    handleResponse,
+    headers: {
```

Comment:
True, but `Content-Type` and `Accept` are probably ones you should always set, so could be overridden. 

Related side note, this is kind of weird:

```
new Headers({foo: undefined}).get('foo')
"undefined"

new Headers({foo: null}).get('foo')
"null"
```

Perhaps if it is `null` or `undefined` we just remove them instead of letting it turned into a string like this.